include/wx/string.h

```diff
@@ -1824,7 +1824,11 @@ class WXDLLIMPEXP_BASE wxString
     if ( !ch.IsAscii() )
         m_impl = wxStringOperations::EncodeChar(ch);
     else
-#endif // wxUSE_UNICODE_UTF8
+#elif wxUSE_UNICODE_WCHAR && SIZEOF_WCHAR_T == 2
+    if ( ch >= 0x10000 )
```

Comment:
Also for readability concerns, could it be worth having an (inline, not to harm performance) `IsUTF16Surrogate()` check instead of using `0x10000` everywhere (notice that in other existing code we also use `<= 0xffff` instead of this test...). I'm actually not sure if it's really going to help, so please feel free to ignore this comment if you think it won't.